flybywiresim/aircraft
# fix(ui): mitigate pop-up camera state bugs

Fixes an issue with pop-ups being unable to be dismissed

## Summary of Changes

- Returns focus back to camera if it has been taken by any instruments.
- Forces Camera substate (Lock/Unlocked) to be unlocked

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub):

## Testing instructions

1. Start a flight.
2. Press Middle Mouse Button (Toggle Freelook), then ALT-TAB while the camera view is still locked to the mouse.
3. Press Ctrl+E To get the engine start-up popup.
4. The mouse cursor should be available and the option to resume can be clicked.
5. Enter PERF page and all MFD/MCDU pages required for T/D to appear on F-PLN
6. Enable pause on T/D in the EFB
7. Get to CRZ flight level
8. Press Middle Mouse Button (Toggle Freelook), then ALT-TAB while the camera view is still locked to the mouse.
9. Wait until Top of Descent
10. See Pause at Top of Descent Pop-up appear.
11. The mouse cursor should be available and the resume button can be clicked.


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
